### ISSUE TITLE
[apex] `pmd:ApexUnitTestClassShouldHaveRunAs` runs per-function instead of per-class, causing a false positive

### ISSUE THREAD

**Issue by stephenlf** [pmd/pmd#6404]
**Affects PMD Version:** v7.19.0, via `sf code-analyzer` core plugin version v5.8.0

**Rule:** `pmd:ApexUnitTestClassShouldHaveRunAs`

Please provide the rule name and a link to the rule documentation:
https://pmd.github.io/pmd/pmd_rules_apex_bestpractices.html#apexunittestclassshouldhaverunas

**Description:** Rule message specifies that there must be at least one `System.runAs` block per test _class_. However, PMD is checking for each test _function_.

**Code Sample demonstrating the issue:** 

```java[apex]
@IsTest
private class MinEx {
    @IsTest
    static void testAccountIdOnInsert() {
        Assert.isTrue(True);
    }

    @IsTest
    static void runAsUser() {
        User user = Utils.newUser();
        System.runAs(am) {
            Test.startTest();
            Test.stopTest();
        }
    }
}
```

**Expected outcome:** 

PMD reports a violation at line 4:12, but that's wrong. That's a false positive. Violation:

```
4 (Low)        pmd:ApexUnitTestClassShouldHaveRunAs     MinEx.cls:4:12    Apex unit test classes should have at least one System.runAs() call
```

**Running PMD through:** *`sf code-analyzer` CLI*

**Comment by dschach** [comment]
I came here to post this.

The test name is wrong. From what I can see, the Java is checking per test method, so it is working as intended... but is not consistent with the test name.

Can we update the name of the rule? Do we need a second rule so things don't break all over the place? Can a rule have two names?
